### PR TITLE
Fixed markdown editor bugs

### PR DESCRIPTION
**Pull request by mistikk**

### CODE REVIEW

File: src/components/markdownEditor/view/formats/applyWebLinkFormat.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default ({ getState, item, setState, isImage = null }) => {
       };
     }
   }
-  setState({ text: newText }, () => {
-    setState({ newSelection });
-  });
+
+  await setState({ text: newText, textUpdated: true });
+  await setState({ newSelection });
```
Comment (ue) [inline src/components/markdownEditor/view/formats/applyWebLinkFormat.js]: await setState({ text: newText }, () => {
 await setState({ newSelection });
});

Comment (ue) [reply src/components/markdownEditor/view/formats/applyWebLinkFormat.js]: its looks better right?

File: src/components/markdownEditor/view/formats/applyWrapFormatNewLines.js [hunk 2]
```diff
@@ -41,15 +41,7 @@ export default ({ getState, item, setState }) => {
...
-    },
-  };
-  setState({ text: newText }, () => {
-    setTimeout(() => {
-      setState({ ...extra });
-    }, 25);
-  });
+
+  await setState({ text: newText, textUpdated: true });
+  await setState({ newSelection: { start: newPosition, end: newPosition } });
```
Comment (ue) [inline src/components/markdownEditor/view/formats/applyWrapFormatNewLines.js]: same thing as above

File: src/components/markdownEditor/view/formats/applyWebLinkFormat.js (hunk 1 above)
Comment (mistikk) [reply src/components/markdownEditor/view/formats/applyWebLinkFormat.js]: I am not sure is it better or same but I can change with your recommended

**Comment by mistikk** [comment]
@ue fixed

**Pull request merged**